plugins/modules/win_domain_ou.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_domain_ou
+short_description: Manage Active Directory Organizational Units
+author: Joe Zollo (@joezollo)
+requirements:
+  - This module requires Windows Server 2012 or Newer
+description:
+  - Manage Active Directory Organizational Units
+  - Adds, Removes and Modifies Active Directory Organizational Units
+  - Task should be delegated to a Windows Active Directory Domain Controller
+options:
+  name:
+    description:
+      - The name of the Organizational Unit
+    type: str
+    required: true
+  protected:
+    description:
+      - Indicates whether to prevent the object from being deleted. When this
+        l(protected=true), you cannot delete the corresponding object without
+        changing the value of the property.
+    type: bool
+    default: false
+  path:
+    description:
+      - Specifies the X.500 path of the OU or container where the new object is
+        created.
+    type: str
+    required: true
+  state:
+    description:
+      - Specifies the desired state of the OU.
+      - When l(state=present) the module will attempt to create the specified
+        OU if it does not already exist.
+      - When l(state=absent), the module will remove the specified OU.
+      - When l(state=absent) and l(recursive=true), the module will remove all
+        the OU and all child OU's.
+    type: str
+    default: present
+    choices: [ present, absent ]
+  recursive:
```

Comment:
I'm torn with this one, we don't need to specify `recursive: yes` for `win_file` when deleting a directory. Was there a particular reason why you you decided to have this as an opt-in behaviour?

Reply:
This was done mainly out of caution - I wanted to be absolutely sure that the intent of the user is to perform a recursive delete operation. I'm not opposed to changing it at all.

Reply:
I'll bring it up in the next windows working group and see what the general consensus is on this.

Reply:
So the windows working group decided on not having this option and to always use the `-Recursive` switch on the remove parameter.

Reply:
@jborean93 Sounds good, i'll kill this option - refactor in progress!

Reply:
Would we want to kill this option or just default it to true?

Reply:
I would say you just set the default to True if that is what the group wanted... though the safe way would be to default to False.  
In large environments with nested OU's deleting everything recursively by default sounds like a bad idea.   but having the option to not do recursive allows the user to make the choice.